### PR TITLE
add first job for rosacli day-2 testing

### PR DESCRIPTION
**Pull request by yuwang-RH**
add first job for rosacli day-2 testing

### CODE REVIEW

**Comment by yuwang-RH** [comment]
@yasun1 @jianzhangbjz @Xia-Zhao-rh Could you help review this PR? Thanks.

**Comment by yuwang-RH** [comment]
@yasun1  I updated with your comment. Could you help review again?

File: ci-operator/step-registry/openshift-extended/rosacli-tests/openshift-extended-rosacli-tests-commands.sh [hunk 1]
```diff
@@ -0,0 +1,263 @@
...
+if [[ ! -z "${ROSA_TOKEN}" ]]; then
+  echo "Logging into ${ROSA_LOGIN_ENV} with offline token"
+  rosa login --env "${ROSA_LOGIN_ENV}" --token "${ROSA_TOKEN}"
+  if [ $? -ne 0 ]; then
+    echo "Login failed"
+    exit 1
+  fi
+else
+  echo "Cannot login! You need to specify the offline token ROSA_TOKEN!"
+  exit 1
```
Comment (yasun1) [inline ci-operator/step-registry/openshift-extended/rosacli-tests/openshift-extended-rosacli-tests-commands.sh:59]: Duplicate logic with line 25

Comment (yuwang-RH) [reply ci-operator/step-registry/openshift-extended/rosacli-tests/openshift-extended-rosacli-tests-commands.sh:59]: @yasun1 Line 25 if export the ROSA_TOKEN. Here is checking if the the env variable exists.

**Comment by yuwang-RH** [comment]
@jianzhangbjz @Xia-Zhao-rh Could you help review and approve this PR? Thanks.

**Comment by yuwang-RH** [comment]
@kuiwang02 Could you help review and approve this PR? Thanks.

**Comment by kuiwang02** [comment]
iamok from golang's view.

**Comment by yuwang-RH** [comment]
@liangxia Could you help review this PR?Thanks.

**Comment by yuwang-RH** [comment]
@jhou1 As @liangxia is coming back at Jul 23, could you help review and approve this PR?

File: ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-master__rosacli.yaml [hunk 2]
```diff
@@ -0,0 +1,37 @@
+base_images:
+  cli:
+    name: "4.14"
```
Comment (jhou1) [inline ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-master__rosacli.yaml:3]: The filename is `openshift-openshift-tests-private-master__rosacli.yaml`, by convention the file name should reflect the version under testing which I suppose is 4.14 for the `cli`  and `tests-private` in base_images are 4.14. Please rename to `openshift-openshift-tests-private-4.14__rosacli.yaml`

**Comment by jhou1** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-master-rosacli-rosa-aws-sts-rosacli-e2e-p2-f3

File: ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-master__rosacli.yaml (hunk 2 above)
Comment (yuwang-RH) [reply ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-master__rosacli.yaml:3]: @jhou1  I share some background here. This job is for the rosacli(rosa command line) testing, not aligned to OCP version, the job will build the rosacli with the latest code as always.  Do you think it is feasible？

**Comment by yuwang-RH** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-master-rosacli-rosa-aws-sts-rosacli-e2e-p2-f3

**Comment by yuwang-RH** [comment]
@yasun1 @kuiwang02 @jhou1 I did some update on openshift-extended-rosacli-commands.sh to add some env variables. all test are passed. The '/lgtm' labels are removed as the new change, could you help review and approve this PR again?

File: ci-operator/step-registry/openshift-extended/rosacli-tests/openshift-extended-rosacli-tests-commands.sh [hunk 3]
```diff
@@ -0,0 +1,267 @@
...
+test -f "${KUBECONFIG}" && (ls -l "${KUBECONFIG}" || true) || echo "kubeconfig file does not exist"
+ls -l ${SHARED_DIR}/kubeconfig || echo "no kubeconfig in shared_dir"
+ls -l ${SHARED_DIR}/kubeadmin-password && echo "kubeadmin passwd exists" || echo "no kubeadmin passwd in shared_dir"
+
+which extended-platform-tests
+
+#set env for rosacli login
+if test -f "${CLUSTER_PROFILE_DIR}/ocm-token"
+then
+    ROSA_TOKEN=$(cat "${CLUSTER_PROFILE_DIR}/ocm-token") || true
```
Comment (kuiwang02) [inline ci-operator/step-registry/openshift-extended/rosacli-tests/openshift-extended-rosacli-tests-commands.sh:31]: @yuwang-RH 
```shell
    ROSA_TOKEN=$(cat "${CLUSTER_PROFILE_DIR}/ocm-token") || true
    export ROSA_TOKEN
```

if `cat "${CLUSTER_PROFILE_DIR}/ocm-token"`, token is not got. you still need to export it?

Comment (yasun1) [reply ci-operator/step-registry/openshift-extended/rosacli-tests/openshift-extended-rosacli-tests-commands.sh:31]: BTW, it is duplicate logic here: `test -f "${CLUSTER_PROFILE_DIR}/ocm-token` and `|| true`.

**Comment by yuwang-RH** [comment]
@yasun1 @kuiwang02 I updated based on your last comment, could you help review and approve this PR again?

**Comment by yuwang-RH** [comment]
@jhou1 Could you help review and approve this PR again? Thanks.

**Pull request merged**